bin/elasticurl/main.cpp

```diff
@@ -0,0 +1,522 @@
+                }
+                else if (!strcmp(aws_cli_optarg, "ERROR"))
+                {
+                    ctx.LogLevel = Aws::Crt::LogLevel::Error;
+                }
+                else
+                {
+                    std::cerr << "unsupported log level " << aws_cli_optarg << std::endl;
+                    s_Usage(1);
+                }
+                break;
+            case 'V':
+                std::cerr << "elasticurl " << ELASTICURL_VERSION << std::endl;
+                exit(0);
+            case 'w':
+                ctx.alpn = "h2";
+                ctx.RequiredHttpVersion = Http::HttpVersion::Http2;
+                break;
+            case 'W':
+                ctx.alpn = "http/1.1";
+                ctx.RequiredHttpVersion = Http::HttpVersion::Http1_1;
+                break;
+            case 'h':
+                s_Usage(0);
+                break;
+            default:
+                std::cerr << "Unknown option\n";
+                s_Usage(1);
+        }
+    }
+
+    if (ctx.InputBody == nullptr)
+    {
+        ctx.InputBody = std::make_shared<std::stringstream>("");
+    }
+
+    if (aws_cli_optind < argc)
+    {
+        struct aws_byte_cursor uri_cursor = aws_byte_cursor_from_c_str(argv[aws_cli_optind++]);
+
+        ctx.uri = Io::Uri(uri_cursor, ctx.allocator);
+        if (ctx.uri.LastError())
+        {
+            std::cerr << "Failed to parse uri" << (char *)uri_cursor.ptr << "with error "
+                      << aws_error_debug_str(ctx.uri.LastError()) << std::endl;
+            s_Usage(1);
+        };
+    }
+    else
+    {
+        std::cerr << "A URI for the request must be supplied.\n";
+        s_Usage(1);
+    }
+}
+
+int main(int argc, char **argv)
+{
+    struct aws_allocator *allocator = aws_default_allocator();
+    (void)allocator;
+    (void)argc;
+    (void)argv;
+
+    struct ElasticurlCtx appCtx;
+    appCtx.allocator = allocator;
+
+    s_ParseOptions(argc, argv, appCtx);
+    ApiHandle apiHandle(allocator);
+    if (appCtx.TraceFile)
+    {
+        apiHandle.InitializeLogging(appCtx.LogLevel, appCtx.TraceFile);
+    }
+    else
+    {
+        apiHandle.InitializeLogging(appCtx.LogLevel, stdout);
+    }
+    bool use_tls = true;
+    uint16_t port = 443;
+    if (!appCtx.uri.GetScheme().len && (appCtx.uri.GetPort() == 80 || appCtx.uri.GetPort() == 8080))
+    {
+        use_tls = false;
+    }
+    else
+    {
+        ByteCursor scheme = appCtx.uri.GetScheme();
+        if (aws_byte_cursor_eq_c_str_ignore_case(&scheme, "http"))
+        {
+            use_tls = false;
+        }
+    }
+
+    auto hostName = appCtx.uri.GetHostName();
+
+    Io::TlsContextOptions tlsCtxOptions;
+    Io::TlsContext tlsContext;
+    Io::TlsConnectionOptions tlsConnectionOptions;
+    if (use_tls)
+    {
+        if (appCtx.cert && appCtx.key)
+        {
+            tlsCtxOptions = Io::TlsContextOptions::InitClientWithMtls(appCtx.cert, appCtx.key);
+        }
+        else
```

Comment:
We got an `#ifdef _WIN32` branch here in c elasticurl. But I cannot find the c++ binding for `aws_tls_ctx_options_init_client_mtls_from_system_path`.